File: code/Cpp/Day1_Second_largest.cpp
#include<climits>

class Solution {
  public:
    int getSecondLargest(vector<int> &arr) {
        
        if(arr.size()<2) return -1;
        
        int first=INT_MIN,second=INT_MIN;
        
        for(int i=0; i<arr.size();i++){
            
            if(arr[i]>first && arr[i]>second) {
                second=first;
                first=arr[i];
            }
            else if(arr[i]>second && arr[i]<first) second=arr[i];
            else continue;
        }
        if (INT_MIN!=second) return second;
        else return -1;
        
    }
};

File: code/Python/Day_1_Second_Largest.py
class Solution:
    def getSecondLargest(self, arr):
        if len(arr)<2:
            return -1
        
        max_element = max(arr)
        
        res_arr = [i for i in arr if i!=max_element]
        
        if res_arr!=[]:
            return max(res_arr)
        else:
            return -1
